### ISSUE TITLE
pip-compile exception UnicodeDecodeError

### ISSUE THREAD

**Issue by hyperknot** [jazzband/pip-tools#439]
Describe the issue briefly here.
If the file contains `pyramid-ipython`, pip-compile termintes with `UnicodeDecodeError`.

##### Steps to replicate

1. `echo pyramid-ipython > requirements.in`
2. `pip-compile`

##### Expected result
no error

##### Actual result
```
Traceback (most recent call last):
  File "/usr/local/bin/pip-compile", line 11, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/piptools/scripts/compile.py", line 185, in cli
    results = resolver.resolve()
  File "/usr/local/lib/python2.7/site-packages/piptools/resolver.py", line 109, in resolve
    has_changed, best_matches = self._resolve_one_round()
  File "/usr/local/lib/python2.7/site-packages/piptools/resolver.py", line 191, in _resolve_one_round
    for best_match in best_matches
  File "/usr/local/lib/python2.7/site-packages/piptools/resolver.py", line 192, in <genexpr>
    for dep in self._iter_dependencies(best_match))
  File "/usr/local/lib/python2.7/site-packages/piptools/resolver.py", line 261, in _iter_dependencies
    dependencies = self.repository.get_dependencies(ireq)
  File "/usr/local/lib/python2.7/site-packages/piptools/repositories/pypi.py", line 135, in get_dependencies
    dependencies = reqset._prepare_file(self.finder, ireq)
  File "/usr/local/lib/python2.7/site-packages/pip/req/req_set.py", line 587, in _prepare_file
    session=self.session, hashes=hashes)
  File "/usr/local/lib/python2.7/site-packages/pip/download.py", line 810, in unpack_url
    hashes=hashes
  File "/usr/local/lib/python2.7/site-packages/pip/download.py", line 653, in unpack_http_url
    unpack_file(from_path, location, content_type, link)
  File "/usr/local/lib/python2.7/site-packages/pip/utils/__init__.py", line 599, in unpack_file
    flatten=not filename.endswith('.whl')
  File "/usr/local/lib/python2.7/site-packages/pip/utils/__init__.py", line 492, in unzip_file
    fn = os.path.join(location, fn)
  File "/usr/local/Cellar/python/2.7.12_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/posixpath.py", line 73, in join
    path += '/' + b
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc6 in position 27: ordinal not in range(128)
```

**Comment by hyperknot** [comment]
10 minutes later and the same command doesn't give exception. Everything is 100% the same. Git reset same.

**Comment by mlazowik** [comment]
Related: #224 #433 #434

**Comment by vphilippon** [comment]
Should be fixed by #568, closing.
